guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-block.md

```diff
@@ -0,0 +1,268 @@
+```bash
+./bin/build-administration.sh
+```
+{% endtab %}
+{% endtabs %}
+
+Note: Your plugin has to be installed and activated. If you still can't see the block make sure to clear your cache as well.
+
+## What about the slots?
+
+If you did everything correct so far, you should see the CMS block in your "Shopping Experiences" module in the category text. You also should be able
+to drag and drop the block into your CMS-Page but you will not see a button there. Instead you will see the exact same "Lorem Ipsum" text like you do
+when you drag and drop a normal text block into your page. Why?
+
+Because of the slot we defined. In our `<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/index.js`. We gave the slot
+the element "text". You can throw in every other element here as well. Since we haven't created a element for the button it's one of our To-Do's now.
+
+We also need a config and we need to react on the settings which are made in the administration to display everything correct in the frontend. First of all
+we need to create a frontend view first. So let's start with that!
+
+## Frontend implemantation
+
+Creating a frontend view is very simple as you should know from the [Add CMS block](../add-cms-block.md) guide. We only need to create a twig file in the 
+correct directory with the same name. It should look like this:
+
+
+{% code title="<plugin root>/src/Resources/views/storefront/block/cms-block-cms-button.html.twig" %}
+{% raw %}
+```text
+{% block block_cms_button_example %}
+    <div class="cms-element ninja-cms-button-flex ninja-cms-button-flex-center">
+		{% block element_text_button %}
+			<a href="#">
+				<button class="ninja-cms-btn">
+                    Button
+                </button>
+			</a>
+		{% endblock %}
+    </div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+Let's do the storefront implementation as previously mentioned in the [Add CMS block](../add-cms-block.md) guide:
+
+{% code title="<plugin root>/src/Resources/views/storefront/block/cms-block-image-text-reversed.html.twig" %}
+{% raw %}
+```text
+{% sw_extends '@Storefront/storefront/block/cms-block-image-text.html.twig' %}
+```
+{% endraw %}
+{% endcode %}
+
+It would be possible to override the code from the [original 'image\_text' file](https://github.com/shopware/platform/blob/v6.3.4.1/src/Storefront/Resources/views/storefront/block/cms-block-image-text.html.twig) file here. We just extend so the code in that file exactly matches 
+the code from the original file.
+
+Now we already should see a button in the storefront if we save a page with our new created CMS block.
+However the view in the administration and in the storefront are completely different. We could simply but our code in the storefront twig file
+and display whatever we want. To really create a button with a config we need to create a matching CMS-Element.
+
+Before we do that we quickly add a bit CSS to our storefront like this:
```

Comment:
```suggestion
It would be possible to override the code from the [original 'image\_text' file](https://github.com/shopware/platform/blob/v6.3.4.1/src/Storefront/Resources/views/storefront/block/cms-block-image-text.html.twig) file here. We extend so the code in that file exactly matches 
the code from the original file.

We already should see a button in the storefront if we save a page with our newly created CMS block.
However, the view in the administration and the storefront is completely different. We could simply put our code in the storefront twig file
and display whatever we want. To really create a button with a config, we need to create a matching CMS-Element.

Before we do that, we quickly add a bit of CSS to our storefront like this:
```